File: app/src/main/java/com/antyzero/smoksmog/time/TimeModule.java
package com.antyzero.smoksmog.time;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public class TimeModule {

    @Provides
    @Singleton
    public CountdownProvider provideCountdownProvider( Context context ) {
        return new CountdownProvider( context );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/time/TimeModule.kt
package com.antyzero.smoksmog.time

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Singleton
class TimeModule {

    @Provides
    @Singleton
    internal fun provideCountdownProvider(context: Context): CountdownProvider {
        return CountdownProvider(context)
    }
}
